hilti/toolchain/include/ast/module.h

```diff
@@ -31,7 +31,7 @@ class Module : public NodeBase {
     const auto& statements() const { return child<statement::Block>(1); }
     const auto& declarations() const {
         if ( _cache.declarations.empty() )
-            _cache.declarations = childs<Declaration>(2, -1);
+            _cache.declarations = childsOfType<Declaration>();
```

Comment:
What's this change needed for?

Reply:
When prune `declaration::Function`s we replace them with `None` nodes. With `childs<Declaration>(2, -1)` we assume that all nodes in `[2, -1]` are `Declaration`s and abort otherwise; the modified form just filters these nodes out.